apps/prairielearn/src/lib/chunks.ts

```diff
@@ -754,73 +752,33 @@ export async function updateChunksForCourse({
  * Generates all chunks for a list of courses.
  */
 export async function generateAllChunksForCourseList(course_ids: string[], authn_user_id: string) {
-  const jobSequenceOptions = {
-    user_id: authn_user_id,
-    authn_user_id: authn_user_id,
+  const serverJob = await createServerJob({
+    userId: authn_user_id,
+    authnUserId: authn_user_id,
     type: 'generate_all_chunks',
     description: 'Generate all chunks for a list of courses',
-  };
-  const job_sequence_id = await serverJobs.createJobSequenceAsync(jobSequenceOptions);
-
-  // don't await this, we want it to run in the background
-  // eslint-disable-next-line no-floating-promise/no-floating-promise
-  _generateAllChunksForCourseListWithJobSequence(course_ids, authn_user_id, job_sequence_id);
-
-  // return immediately, while the generation is still running
-  return job_sequence_id;
-}
+  });
 
-/**
- * Helper function to actually generate all chunks for a list of courses.
- */
-async function _generateAllChunksForCourseListWithJobSequence(
-  course_ids: string[],
-  authn_user_id: string,
-  job_sequence_id: string
-) {
-  try {
-    for (let i = 0; i < course_ids.length; i++) {
-      const course_id = course_ids[i];
-      const jobOptions = {
-        course_id: null /* Set the job's course_id to null so we can find it from the admin page */,
-        type: 'generate_all_chunks',
-        description: `Generate all chunks for course ID = ${course_id}`,
-        job_sequence_id,
-        user_id: authn_user_id,
-        authn_user_id,
-        last_in_sequence: i === course_ids.length - 1,
-      };
-      const job = await serverJobs.createJobAsync(jobOptions);
-      job.info(chalkDim(`Course ID = ${course_id}`));
-
-      try {
-        await _generateAllChunksForCourseWithJob(course_id, job);
-        job.succeed();
-      } catch (err) {
-        job.error(chalk.red(JSON.stringify(err)));
-        await job.failAsync(err);
-        throw err;
-      }
+  serverJob.executeInBackground(async (job) => {
+    for (const [i, courseId] of course_ids.entries()) {
+      job.info(`Generating chunks for course ${courseId} [${i + 1}/${course_ids.length}]`);
+      await _generateAllChunksForCourseWithJob(courseId, job);
```

Comment:
Still loving how much more concise new-style job code is!

Reply:
Sooooooo much better!